packages/grid/x-data-grid-premium/src/hooks/features/rowGrouping/useGridRowGrouping.tsx

```diff
@@ -34,13 +33,21 @@ import { GridInitialStatePremium } from '../../../models/gridStatePremium';
 
 export const rowGroupingStateInitializer: GridStateInitializer<
   Pick<DataGridPremiumProcessedProps, 'rowGroupingModel' | 'initialState'>
-> = (state, props) => ({
-  ...state,
-  rowGrouping: {
-    model: props.rowGroupingModel ?? props.initialState?.rowGrouping?.model ?? [],
-    unstable_sanitizedModelOnLastRowTreeCreation: [],
-  },
-});
+> = (state, props, apiRef) => {
+  apiRef.current.unstable_caches = {
```

Comment:
For the rows state initializer we don't pass a new object, but update directly the key.

https://github.com/mui/mui-x/blob/23a57c3665ff00675c976cc89f8aaeee54cefe69/packages/grid/x-data-grid/src/hooks/features/rows/useGridRows.ts#L33